src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerService.cs

```diff
@@ -120,6 +121,39 @@ public void InitializeService(ServiceHost serviceHost)
             this.SessionMonitor.AddSessionListener(this);
         }
 
+        /// <summary>
+        /// Handle request to start a profiling session
+        /// </summary>
+        internal async Task HandleStartProfilingRequest(StartProfilingParams parameters, RequestContext<StartProfilingResult> requestContext)
+        {
+            await Task.Run(async () =>
+            {
+                try
+                {
+                    ConnectionInfo connInfo;
+                    ConnectionServiceInstance.TryFindConnection(
+                        parameters.OwnerUri,
+                        out connInfo);
+                    if (connInfo != null)
+                    {
+                        // create a new XEvent session and Profiler session
+                        var xeSession = this.XEventSessionFactory.GetXEventSession(parameters.SessionName, connInfo);
+                        monitor.StartMonitoringSession(parameters.OwnerUri, xeSession);
+                        var result = new StartProfilingResult();
+                        await requestContext.SendResult(result);
+                    }
+                    else 
+                    {
+                        throw new Exception(SR.ConnInfoIsNull);
+                    }
+                }
+                catch (Exception e)
+                {
+                    await requestContext.SendError(new Exception(SR.CreateSessionFailed(e.Message)));
```

Comment:
You should add in the original exception as well since it contains all the useful information like callstack

Reply:
yes, let's just do one sendError call.